actors/miner/src/deadline_state.rs

```diff
@@ -366,23 +358,13 @@ impl Deadline {
             .map_err(|_| actor_error!(illegal_state; "partition index out of bitfield range"))?;
         self.early_terminations |= &new_early_terminations;
 
-        let all_on_time_sectors = BitField::union(&on_time_sectors);
-        let all_early_sectors = BitField::union(&early_sectors);
-
         // Update live sector count.
-        let on_time_count = all_on_time_sectors.len();
-        let early_count = all_early_sectors.len();
+        let on_time_count = all_expirations.on_time_sectors.len();
+        let early_count = all_expirations.faulty_sectors.len();
         self.live_sectors -= on_time_count + early_count;
```

Comment:
Why don't we subtract the proof expiring sectors in the expiration set as part of the live_sectors decrease?

Reply:
It probably is a bug missed by tests.